app/models/page.rb

```diff
@@ -2,12 +2,17 @@
 class Page < ActiveRecord::Base
   # Associations
   belongs_to :council
-  has_many :page_elements
+  has_many :page_elements, dependent: :destroy
+  has_many :page_images, dependent: :destroy
 
   # Validations
-  validates :url, uniqueness: true, if: 'url.present?'
+  validates :title, presence: true
+  validates :url, uniqueness: true,
+                  presence: true,
+                  format: { with: /\A[a-z0-9_-]+\z/ },
+                  unless: 'council_id.present?'
```

Comment:
Hehe, rails uses Kernel#eval in validations? That's just evil. Don't they know about Procs?